src/main/java/org/variantsync/diffdetective/diff/difftree/DiffTree.java

```diff
@@ -111,6 +118,33 @@ public static DiffResult<DiffTree> fromDiff(final String diff, boolean collapseM
         return tree;
     }
 
+    public static Result<DiffTree, List<DiffError>> fromPatch(final GitPatch patchInfo, final Repository repository) throws IOException {
+        final CommitDiffResult result = CommitDiffResult.fromCommitInRepository(patchInfo.getCommitHash(), repository);
+        final Path changedFile = Path.of(patchInfo.getFileName());
```

Comment:
Uhhm ok. I though using `Path` wherever possible is the most safe way. For example, if `patchInfo.getFileName()` returns a Windows style path as string but the parser returns a Unix path as string, checking for string equality is insufficient because the paths are different. And indeed we are concerned in comparing paths and not strings here.

Also this path variable is not interacting with Git here and I hope that the JGit library interacts with git in a safe way with respect to paths. If `Path` is the idiomatic way to go in Java, we should probably stick to that.

Reply:
I see. You're probably right. I checked the JGit tests and they assume that the path always uses `'/'` as separator. The windows parser for `Parth.of` in OpenJDK handles such paths like UNIX paths so this should work. Although this might not be standard compatible because the Java documentation explicitly says that this is implementation defined behavior, it's still the easiest way to get it working on most (hopefully all) platforms.